### ISSUE TITLE
Voice search again not working

### ISSUE THREAD

**Issue by y-lee** [yuliskov/SmartYouTubeTV#641]
Looks like a recurrence of #602. No mic button in Search, and when I bring up in-app voice search as per #608, the mic icon vibrates with my speech but nothing happens. Mic works with Google Assistant.

**Comment by BrettAlan128** [comment]
Yep same problem. Does this just happen out of the blue or is the app auto updating? This happened a few weeks ago and it lasted for a while. I would like to just stick to my working version next time a stable version gets released

**Comment by y-lee** [comment]
This issue occurred this time and the last time without any client-side changes, so I don't think there is much that we Smart YouTube users can do. The issue may again fix itself without any client-side changes, too, although perhaps there is something that can be changed in Smart YouTube to avoid another recurrence.

**Comment by albino1** [comment]
Edit: Never mind.

**Comment by fgl27** [comment]
> @fgl27 says in this post that voice search gets disabled when animations are disabled:
> 
> [#634 (comment)](https://github.com/yuliskov/SmartYouTubeTV/issues/634#issuecomment-680278817)
> 
> They also say that @yuliskov disabled animations by default and removed the option to enable them, which means you can't turn voice search back on.

No I didn't wrote that. And this issue has no relation to what I was discussing there. Plus there is no relation regarding animation or voice search.

So you are all wrong on what you say.... Don't miss quote others. That is disrespectful.

**Comment by albino1** [comment]
I didn't mean to be disrespectful, I thought that's what you were saying.  I'll edit my post.

**Comment by y-lee** [comment]
Voice search again working with 6.17.724 (Pro Alt) 320 dpi / SHIELD Android TV (darcy). I downloaded it manually since Smart YouTube didn't prompt me to update.

I have noticed that when issues like this bug occurs, the version and device may say something else than the above (I've seen Samsung mentioned). Does this give you a clue on the cause?

**Comment by y-lee** [comment]
The issue reoccurred once. I wish I'd saved the screenshot, but I do remember that Version stated "webclient" and "Device" mentioned Philips, instead of what is correct for my Shield TV. It has not happened again after I restarted Smart YouTube and I have not been able to reproduce the issue.

**Comment by fgl27** [comment]
@y-lee this issue only occurs when the about shows the wrong info?

I notice that lately sometimes when I open the app it doesn't load the **SmartYouTubeTV** changes, meaning the app loads as the default youtube app that today loads as if was a Philips tv

https://github.com/yuliskov/SmartYouTubeTV/blob/9f92e3b84f0f1cdc29c0b3ad36533ed2f4804d74/common/src/main/java/com/liskovsoft/smartyoutubetv/misc/UserAgentManager.java#L40-L44

When that happens I don't use the app I just close it, some times is necessary to force stop the app, so it opens the correct way.

On that odd case if you go to the about of the app it will show the wrong info (wrong for **SmartYouTubeTV**).

Understanding when the problem happens is necessary to fix the issue, if is only on that case, what is necessary to fix here is to prevent the app from loading without the modifications that make this **SmartYouTubeTV**.

**Comment by fgl27** [comment]
BTW the user agent here is not the problem, regarding the problem that sometimes the app loads without the modifications that make this **SmartYouTubeTV**, I tested with `SAMSUNG_SMART_TV[1]` and it occurs...

It also happens in others devices then the SHIELD...

**Comment by y-lee** [comment]
Voice search did not work for the entire time from when I reported this bug—18 days ago—until today, with many client/Shield restarts along the way, so whatever was the cause wasn't transient. I cannot say whether the wrong info always appeared during that time.

**Comment by fgl27** [comment]
You better test for it then. Only with the correct info the issue can be addressed guesses will not do it.

**Comment by fgl27** [comment]
Here I'm using samsung user agent not philips.

when the app "miss open" doesn't load the **SmartYouTubeTV** changes, the app opens like this

![Smart YouTube TV_20200920_190452](https://user-images.githubusercontent.com/13755892/93723507-61df7000-fb75-11ea-81dc-09050950b63e.png)

I have my to open on the sub screen, it remembers that but I have to click on sing in to get my user to load, but when that happens the app is all mess up so I just close it...

the about of the app show this

![Smart YouTube TV_20200920_190509](https://user-images.githubusercontent.com/13755892/93723528-8f2c1e00-fb75-11ea-8d85-414f8f2bc486.png)


VS the OK load of the app

![Smart YouTube TV_20200920_191022](https://user-images.githubusercontent.com/13755892/93723532-99e6b300-fb75-11ea-88d0-003c4ebc8c9e.png)

**Comment by fgl27** [comment]
And when that issue happens the voice search will not work, but that is expected as the app didn't properly loaded...

There is other issues that occurs, but what is necessary to be fixed here is the loading issue not the voice or wherever other issue there may be, as when the app properly loads the voice search is all fine at least for me.

So if anyone has a different behavior of the voice thing, when the app loads OK share all the necessary info...

Making sure you look in the App version to see if the app really loaded OK, as if it didn't load OK the issue ins't voice but the miss loading.

**Comment by y-lee** [comment]
![Screenshot_20200925-131128](https://user-images.githubusercontent.com/2789219/94312027-12ac8d00-ff31-11ea-8a80-10a167f6240a.png)

Restarting Smart YouTube fixed above issue.

**Comment by fgl27** [comment]
@y-lee also add the info related to when the app load OK (the app version show the version of **SmartYouTubeTV**), if on that case have you notice the voice stop working lately? or if is just on the case of the app miss loading (as on the image you just shared)? 

@yuliskov when you have some time please check the problem of the app not loading properly see the above images, also the problem of the user agent and animations reported here https://github.com/yuliskov/SmartYouTubeTV/issues/654 thanks

**Comment by y-lee** [comment]
@fgl27 I already provided the info when the app loads correctly: 6.17.724 (Pro Alt) 320 dpi / SHIELD Android TV (darcy)

**Comment by fgl27** [comment]
> @fgl27 I already provided the info when the app loads correctly: 6.17.724 (Pro Alt) 320 dpi / SHIELD Android TV (darcy)

You didn't, that is why I made two questions on my last post directed to you, but you didn't answer any of my questions. If I made those questions is because they are direct related to fix the issue.

Any way if you care to fix the issue you started is recommended to answer it.

**Comment by y-lee** [comment]
@fgl27 I don't know what other questions I'm to answer. I've provided all data I have on this issue.

* This issue occurred for 18 days (when I opened the ticket) through multiple client/Shield TV restarts, before stopping seven days ago.
* Since the end of the 18-day period when this issue always occurred, this issue has occasionally reoccurred; when it has reoccurred, the device/client reports "Philips" as per my screenshot.
* Since the end of the 18-day period, restarting the client has always fixed the issue, with the device/client reporting Shield as it is supposed to.

**Comment by fgl27** [comment]
> @fgl27 I don't know what other questions I'm to answer. I've provided all data I have on this issue.
> 
> * This issue occurred for 18 days (when I opened the ticket) through multiple client/Shield TV restarts, before stopping seven days ago.
> * Since the end of the 18-day period when this issue always occurred, this issue has occasionally reoccurred; when it has reoccurred, the device/client reports "Philips" as per my screenshot.
> * Since the end of the 18-day period, restarting the client has always fixed the issue, with the device/client reporting Shield as it is supposed to.

that answers it, the issue is not at least today the voice, but the issue is the app not properly loading.

So technically is better to open a separated issue and close this, or at least, update the title, and edit the first post, including the new information that you have found of what the problem is.

**Comment by y-lee** [comment]
As #663 mentions, YouTube has changed something on the backend again. Here's what I see on Nvidia Shield TV:

![Screenshot_20201009-203640](https://user-images.githubusercontent.com/2789219/95645014-cc4a4880-0a6f-11eb-81f8-efdb3315b2d0.png)

Interestingly, the change happened while I was using Smart YouTube. One moment I could play videos; the next moment I could not.

Restarting the client several times gave the above version and device until finally I was able to again get the correct results one time. After quitting, restarting the client several times was again required before I again got the correct results.
